cmd/protoc-gen-fastmarshal/templates/fieldsnippets.tmpl

```diff
@@ -855,6 +855,62 @@
             m.{{.GoName}} = {{if eq $syntax "proto2"}}csproto.Int{{$bitSize}}(v){{else}}v{{end}}
         }
     {{- end -}}
+{{- else if eq $kind "float" -}}
+    {{- if eq $cardinality "repeated" -}}
+        switch wt {
+        case csproto.WireTypeVarint:
```

Comment:
> I couldn't get repeated fields to work properly,
I think the problem is here.  `float` fields are encoded as `WireTypeFixed32`, not `WireTypeVarint`.  Same this below for `double` fields below (`WireTypeFixed64` instead of `WireTypeVarint`).